opensds/nbp
# [CSI] OFFLINE volume expansion support

What this PR does / why we need it:
This change addresses the offline volume expansion support from side

When the volume demand (pvc) is edited to request more volume csi containers will trigger csi block plugin to provision volume expansion.
Some key highlights

    Node and controller capabilities are extended to acknowledge offline volume support
    csi resizer sidecar container will be started during the deployment to receive expand volume related notification
    Controller Expand Volume expand function is added to trigger the volume expansion at opensds/backend side
    Node Expand Volume function is added to expand the filesystem already mounted on node

Which issue this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close that issue when PR gets merged): fixes #
This PR addresses part(offline) of the issue: #209

Special notes for your reviewer:

    Any other PR(s) this PR is dependant on: opensds/opensds#1174

Test steps:

    Create a pod which uses pvc of size x gb
    Delete just the pod where pvc is still there
    Edit the pvc with size more than initial size
    Create the pod again with the same pvc
    Verify that pv/pvc/opensds volumes all are expanded
    Verify that the volume mounted on node has expanded size

Release note:
